### PR TITLE
Include rocksdb into shaded client jar

### PR DESCRIPTION
**Pull request by beinan**
### What changes are proposed in this pull request?

Include rocksdb into shaded client jar

### Why are the changes needed?

Fix no class found exception when using rocksdb storage of the local cache

### Does this PR introduce any user facing changes?

None

### CODE REVIEW

**Comment by beinan** [comment]
Hello @JySongWithZhangCe , could you help give this a shot? thx!

**Comment by JySongWithZhangCe** [comment]
> Hello @JySongWithZhangCe , could you help give this a shot? thx!

Sure~

File: shaded/client/pom.xml [hunk 1]
```diff
@@ -283,6 +282,7 @@
                     <exclude>org/w3c/dom/**/*</exclude>
                     <exclude>org/xml/sax/*</exclude>
                     <exclude>org/xml/sax/**/*</exclude>
+                    <exclude>org/rocksdb/**/*</exclude>
```
Comment (JySongWithZhangCe) [inline shaded/client/pom.xml:285]: Rocksdb package is not belong to JDK. Maybe we could move `<exclude>org/rocksdb/**/*</exclude>` to another place?

**Comment by JySongWithZhangCe** [comment]
@beinan I have tested, local cache in ROCKS works well. Thx:)

**Comment by ggezer** [comment]
@beinan could you add a comment stating the size difference in client jar with this change?

**Comment by beinan** [comment]
> @beinan could you add a comment stating the size difference in client jar with this change?

Thank you @ggezer ! Good call, we had some discussion on another PR.  We found the size diff as below,

Jar size got increased: from 31M to 70M
31938699 Jan 24 16:18 alluxio-shaded-client-2.8.0-SNAPSHOT.jar
70450298 Jan 26 13:32 alluxio-shaded-client-2.8.0-SNAPSHOT.jar

File: shaded/client/pom.xml (hunk 1 above)
Comment (beinan) [reply shaded/client/pom.xml:286]: Thank you @JySongWithZhangCe !  good call!  just added a comment to make it looks like as a separate section

Comment (JySongWithZhangCe) [reply shaded/client/pom.xml:286]: LGTM！Thinks for your contribution!

**Comment by beinan** [comment]
@ggezer @yuzhu @beinan @jiacheliu3 @rongrong any objections or concerns about including the rocksdb-jni into our shaded client jar?  Jar size got increased from 31M to 70M.  Also, it's not a perfect shading due to the native code which might cause potential conflict if there has been a rocksdb-jni dependency included in the client side app already

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by beinan** [comment]
Tested with spark, the word counting example is running well.

**Comment by yohengyang** [comment]
@beinan

**Pull request merged**